Pester.psm1

```diff
@@ -659,6 +659,48 @@ function Get-ScriptBlockScope
     [scriptblock].GetProperty('SessionStateInternal', $flags).GetValue($ScriptBlock, $null)
 }
 
+function Get-OperatingSystem
```

Comment:
I chose to create a Get-OperatingSystem function to lay the groundwork for other OSes. With a single Get-OperatingSystem function, you can add and remove OSes at will and bundle up all the logic inside there instead of creating different Test-OS functions.

Also, PS core works on MacOS and Linux and it's quite possible that the temp location (that this function was created for) will change amongst MacOS or different Linux distros.